src/Abstractions/NexusMods.Abstractions.Collections/Json/ModSource.cs

```diff
@@ -0,0 +1,20 @@
+using System.Text.Json.Serialization;
+using NexusMods.Abstractions.NexusWebApi.Types;
+using NexusMods.Paths;
+
+namespace NexusMods.Abstractions.Collections.Json;
+
+public class ModSource
+{
+    [JsonPropertyName("type")]
+    public string Type { get; init; } = string.Empty;
```

Comment:
This would probably be better represented as an enum for the purpose of better discovery (of available options)
and use of 'switch' statements.

System.Text.Json has a string to enum converter (`JsonStringEnumConverter`); which we already use in `MembershipRole`.